### PR TITLE
Added RDS Aurora MySQL blue/green deployment resource and test (testing WIP)

### LINKED ISSUE #28956: AWS Aurora Blue/Green Update
### Description

`blue_green_update` is valid only for `aws_db_instance` and does not work with `aws_rds_cluster_instance` . As per the [documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster_instance) `aws_rds_cluster_instance` is the recommended way to manage aurora instance. If that is the case, `blue_green_update` should work with Auroa. (aws_rds_cluster_instance)

### Affected Resource(s) and/or Data Source(s)

aws_rds_cluster, aws_rds_cluster_instance

### Potential Terraform Configuration

_No response_

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by autotune**
### Description
```console
resource "aws_rds_cluster" "default" {
  cluster_identifier = "aurora-cluster-demo-3"
  availability_zones = ["us-east-1b", "us-east-1c", "us-east-1d"]
  database_name      = "mydb"
  master_username    = "foo"
  master_password    = "barbut8chars"
  engine             = "aurora-mysql"
  engine_version     = "5.7.mysql_aurora.2.12.0"
  skip_final_snapshot = true 
  db_cluster_parameter_group_name = "blue57"
}

resource "aws_rds_cluster_instance" "cluster_instances" {
  depends_on         = [aws_rds_cluster.default]
  count              =  2 
  identifier         = "aurora-cluster-${count.index}"
  cluster_identifier = "aurora-cluster-demo-3" 
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
  instance_class     = "db.t3.small"
  db_parameter_group_name = "blue57instance"
}

resource "aws_rds_cluster_blue_green_deployment" "status" {
  cluster_identifier = aws_rds_cluster.default.cluster_identifier
  cleanup_resources = true 
  engine = aws_rds_cluster.default.engine
  switchover_enabled = var.switchover_enabled 
  create_deployment = var.create_deployment
}

```

### Relations
Closes #0000
--->

Closes #28956

### Output from Acceptance Testing
WIP

```console
% make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

**Comment by autotune** [comment]
@justinretzolk I have not been able to build a test for this as the parameter group is unable to wait for Aurora DB instance creation. Going to submit what I've finished so far today, if someone else wants to take over and finish building out the test portion that is fine with me.

File: internal/service/rds/cluster_blue_green_resource.go [hunk 1]
```diff
@@ -0,0 +1,1482 @@
...
+
+							_, err = connv2.DeleteBlueGreenDeployment(ctx, bginput)
+
+							if err != nil {
+								log.Printf("[DEBUG] updating RDS DB Cluster (%s): deleting Blue/Green Deployment source: %s", aws.StringValue(&bluegreen[0]), err)
+							}
+
+						}
+						return true, err
+					},
```
Comment (timclifford) [inline internal/service/rds/cluster_blue_green_resource.go:979]: internal/service/rds/cluster_blue_green_resource.go:979:7: syntax error: unexpected comma at end of statement

File: internal/service/rds/cluster_blue_green_resource.go [hunk 2]
```diff
@@ -0,0 +1,1482 @@
...
+				)
+			}
+		}
+	}()
```
Comment (timclifford) [inline internal/service/rds/cluster_blue_green_resource.go:983]: internal/service/rds/cluster_blue_green_resource.go:983:3: syntax error: unexpected ( after top level declaration

**Comment by bdeetz** [comment]
I'm going to look into the testing portion.

File: internal/service/rds/cluster_blue_green_resource.go (hunk 2 above)
Comment (bdeetz) [reply internal/service/rds/cluster_blue_green_resource.go:983]: Hey @autotune,
In this section of code, there is some funky bracing and indentation issues preventing us from getting past linting to build. I'm trying to debug this issue, but when I fix the braces and indentation I find references to variables that don't exist. Can you fill us in on what you think is going on in this code block?

**Comment by gdavison** [comment]
Thanks for the work you've put into this, @autotune. Unfortunately, a stand-alone resource for RDS Blue/Green Deployments does not fit with the Terraform model of declarative infrastructure: Using the resource would require multiple iterations of editing the Terraform configuration, applying the configuration, or importing resources. A separate tool that can manage the orchestration steps would be a better fit.

Blue/Green Deployments work with `aws_db_instance` because it is a single self-contained resource, whereas an Aurora cluster is made up of one `aws_rds_cluster` and at least one `aws_rds_cluster_instance`.

**Pull request closed without merging**